app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextActions.java

```diff
@@ -10,70 +10,78 @@
 package net.gsantner.markor.format.markdown;
 
 import android.app.Activity;
+import android.support.annotation.StringRes;
 import android.view.HapticFeedbackConstants;
 import android.view.KeyEvent;
 import android.view.View;
-import android.view.ViewGroup;
 
 import net.gsantner.markor.R;
+import net.gsantner.markor.activity.ActionItem;
 import net.gsantner.markor.format.general.CommonTextActions;
 import net.gsantner.markor.model.Document;
 import net.gsantner.markor.ui.AttachImageOrLinkDialog;
 import net.gsantner.markor.ui.SearchOrCustomTextDialogCreator;
 import net.gsantner.markor.ui.hleditor.TextActions;
 import net.gsantner.opoc.util.Callback;
 
+import java.util.Arrays;
+import java.util.List;
+
 public class MarkdownTextActions extends TextActions {
 
     public MarkdownTextActions(Activity activity, Document document) {
         super(activity, document);
     }
 
     @Override
-    public void appendTextActionsToBar(ViewGroup barLayout) {
-        if (barLayout.getChildCount() == 0) {
-            setBarVisible(barLayout, true);
-            for (int[] actions : TMA_ACTIONS) {
-                MarkdownTextActionsImpl actionCallback = new MarkdownTextActionsImpl(actions[0]);
-                appendTextActionToBar(barLayout, actions[1], actions[2], actionCallback, actionCallback);
-            }
-        }
+    public boolean runAction(String action, boolean modLongClick, String anotherArg) {
+        return runCommonTextAction(action);
     }
 
     @Override
-    public boolean runAction(String action, boolean modLongClick, String anotherArg) {
-        return runCommonTextAction(action);
+    protected ActionCallback getActionCallback(@StringRes int keyId) {
+        return new MarkdownTextActionsImpl(keyId);
+    }
+
+    @Override
+    public List<ActionItem> getActiveActionList() {
+        return Arrays.asList(TMA_ACTIONS);
+    }
+
+    @Override
+    protected @StringRes int getFormatActionsKey() {
+        return R.string.pref_key__markdown__action_keys;
```

Comment:
Lots of logic moved to TextActions. By overriding these three 1 line functions, the derived classes can setup the required behavior.